apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ApiModelFactory.java

```diff
@@ -351,9 +350,6 @@ public static IApiBaseline newApiBaselineFromTarget(String name, ITargetDefiniti
 	private static String generateTargetLocation(ITargetDefinition definition) {
 		StringBuilder sb = new StringBuilder(TARGET_PREFIX);
 		sb.append(IPath.SEPARATOR);
-		if (definition instanceof TargetDefinition target) {
-			sb.append(target.getSequenceNumber());
```

Comment:
Could this be the cause of the new test failures? I.e. that it's somehow expected to get different identifier?

Reply:
Yes API tools require the sequence number for some "stale" check but it is only used in the UI (and not even reliable as the sequence number is never persisted...)

Reply:
Understand. Couldn't then API-tools then reuse the `P2TargetUtils.checkProfile()`?

Reply:
The profile check is only one aspect and only valid for IU locations.

Reply:
Yes. When looking at the removed calls of `incrementSequenceNumber()` I can only see the methods
- `setTargetLocations()`
- `setImplicitDependencies()`

that are currently not covered by leveraging `P2TargetUtils.checkProfile()`.

Reply:
I think for API tools it would be more useful to store a hash of the target content, but lets do it step-by step here...